src/test/scala/chiselTests/AnalogSpec.scala

```diff
@@ -0,0 +1,211 @@
+// See LICENSE for license details.
+
+package chiselTests
+
+import chisel3._
+import chisel3.util._
+import chisel3.testers.BasicTester
+import chisel3.experimental.{Analog, attach}
+
+// IO for Modules that just connect bus to out
+class AnalogReaderIO extends Bundle {
+  val bus = Analog(32.W)
+  val out = Output(UInt(32.W))
+}
+// IO for Modules that drive bus from in (there should be only 1)
+class AnalogWriterIO extends Bundle {
+  val bus = Analog(32.W)
+  val in = Input(UInt(32.W))
+}
+
+trait AnalogReader {
+  self: Module =>
+  final val io = self.IO(new AnalogReaderIO)
+}
+
+class AnalogReaderBlackBox extends BlackBox with AnalogReader
+
+class AnalogReaderWrapper extends Module with AnalogReader {
+  val mod = Module(new AnalogReaderBlackBox)
+  io <> mod.io
+}
+class AnalogWriterBlackBox extends BlackBox {
+  val io = IO(new AnalogWriterIO)
+}
+// Connects two Analog ports
+class AnalogConnector extends Module {
+  val io = IO(new Bundle {
+    val bus1 = Analog(32.W)
+    val bus2 = Analog(32.W)
+  })
+  io.bus1 <> io.bus2
+}
+
+// Parent class for tests connecing up AnalogReaders and AnalogWriters
+abstract class AnalogTester extends BasicTester {
+  final val BusValue = "hdeadbeef".U
+
+  final val (cycle, done) = Counter(true.B, 2)
+  when (done) { stop() }
+
+  final val writer = Module(new AnalogWriterBlackBox)
+  writer.io.in := BusValue
+
+  final def check(reader: Module with AnalogReader): Unit =
+    assert(reader.io.out === BusValue)
+}
+
+class AnalogSpec extends ChiselFlatSpec {
+  behavior of "Analog"
+
+  it should "NOT be bindable to registers" in {
+    a [ChiselException] should be thrownBy {
+      elaborate { new Module {
+        val io = IO(new Bundle {})
+        val reg = Reg(Analog(32.W))
+      }}
+    }
+  }
+
+  it should "NOT be bindable to a direction" in {
+    a [ChiselException] should be thrownBy {
+      elaborate { new Module {
+        val io = IO(new Bundle {
+          val a = Input(Analog(32.W))
+        })
+      }}
+    }
+    a [ChiselException] should be thrownBy {
+      elaborate { new Module {
+        val io = IO(new Bundle {
+          val a = Output(Analog(32.W))
+        })
+      }}
+    }
+  }
+
+  it should "be flippable" in {
+    elaborate { new Module {
+      val io = IO(new Bundle {
+        val a = Flipped(Analog(32.W))
+      })
+    }}
+  }
+
+  // There is no binding on the type of a memory
+  // Should this be an error?
+  ignore should "NOT be a legal type for Mem" in {
+    a [ChiselException] should be thrownBy {
+      elaborate { new Module {
+        val io = IO(new Bundle {})
+        val mem = Mem(16, Analog(32.W))
+      }}
+    }
+  }
+
+  it should "NOT be bindable to Mem ports" in {
```

Comment:
Practically this is equivalent to the above, both because it's going to fail at the Mem creation stage, and that if a Analog Mem is allowed then so should a Analog Mem port.